src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -8637,18 +8673,34 @@ public ValueTuple(T1 item1, T2 item2)
         }
     }
 }";
+            var expected = new[]
+            {
+                    // (13,20): warning CS0169: The field '(T1, T2).Item2' is never used
+                    //         private T2 Item2;
+                    Diagnostic(ErrorCode.WRN_UnreferencedField, "Item2").WithArguments("(T1, T2).Item2").WithLocation(13, 20),
+                    // (14,16): error CS0171: Field '(T1, T2).Item2' must be fully assigned before control is returned to the caller
```

Comment:
>error CS0171: Field '(T1, T2).Item2' must be fully assigned before control is returned to the caller [](start = 32, length = 100)

Are these errors because there's no `public Item2`? (The errors are confusing, but not necessarily a blocking issue.)

Reply:
The symbol for ValueTuple has two Item2 members.
One is the private one which appears in source and the other one is the error symbol which is added to tuples who lack it.
That's causing the confusing diagnostics.
I would rather leave as-is given that we're talking about diagnostic quality in a dark corner (someone making their own ValueTuple type and doing it badly).

---
In reply to: [356307468](https://github.com/dotnet/roslyn/pull/39370#discussion_r356307468) [](ancestors = 356307468)